gpii/node_modules/gpii-oauth2/gpii-oauth2-authz-server/public/src/core/js/GPIIOauth2.js

```diff
@@ -0,0 +1,32 @@
+/*!
+GPII OAuth2 server
+
+Copyright 2015 OCAD University
+
+Licensed under the New BSD license. You may not use this file except in
+compliance with this License.
+
+You may obtain a copy of the License at
+https://github.com/gpii/universal/LICENSE.txt
+*/
+
+/* global fluid, jQuery */
+
+var gpii = gpii || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    fluid.registerNamespace("gpii.oauth2");
+
+    gpii.oauth2.ajax = function (urlTemplate, urlParams, options) {
+        var url = fluid.stringTemplate(urlTemplate, urlParams);
```

Comment:
I have changed gpii.oauth2.ajax from the version in https://github.com/simonbates/universal/pull/12

https://github.com/simonbates/universal/commit/14164be56154b6711d84b90d58cb47672a5fd16e

The initial fix was to call encodeURI() on the result of fluid.stringTemplate(). This encoding will be too late for values which contain characters with special meanings in the URI query part such as "?" and "&" -- encodeURI() will leave them as is.

I have changed the code to instead call encodeURIComponent() on each individual replacement value before insertion into the template.